common/go-utils/src/inv.rs

```diff
@@ -137,7 +151,36 @@ pub fn list_upstream_artifacts() -> Result<Vec<Artifact>, ListUpstreamArtifactsE
         .map_err(ListUpstreamArtifactsError::ParseJsonResponse)?
         .iter()
         .flat_map(|release| &release.files)
-        .filter(|file| !file.sha256.is_empty() && file.os == "linux" && file.arch == "amd64")
+        .filter(|file| {
+            !file.sha256.is_empty()
+                && file.os == "linux"
+                && (file.arch == "amd64" || file.arch == "arm64")
+        })
         .map(|file| Artifact::try_from(file).map_err(ListUpstreamArtifactsError::Conversion))
         .collect::<Result<Vec<_>, _>>()
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_artifact_display_format() {
+        let artifact = Artifact {
+            go_version: "go1.7.2".to_string(),
+            semantic_version: Version::parse("1.7.2").unwrap(),
+            os: "linux".to_string(),
+            arch: "amd64".to_string(),
+            url: String::new(),
+            sha_checksum: String::new(),
+        };
+
+        assert_eq!(
+            format!(
+                "{} ({}-{})",
+                artifact.go_version, artifact.os, artifact.arch
+            ),
```

Comment:
since you control the inputs you should just check the expected value instead of duplicating the implementation 😅 

```suggestion
            "go1.7.2 (linux-x86_64)",
```

Reply:
Isn't the *current* implementation primarily what we want to test (to ensure that future changes to the implementation is caught by this test)?

Happy to change this, but it seems testing against `"go1.7.2 (linux-x86_64)"` would end up testing for unrelated implementation details (e.g. how `ToString`/`Display` is implemented for `go_version`/`os`/`arch`)? The latter two, `os` and `arch` have been changed to enums since you reviewed this, and each now implement the `Display` trait and include tests for the expected behavior.

Reply:
yeah, this suggestion is now muddied up by the new `enum` implementations added for `os` and `arch`.  i suppose you could disregard.

Reply:
I changed it to test for actual expected value anyway in https://github.com/heroku/buildpacks-go/pull/230/commits/847f9389fd1bf3d8ad7c22bad7100c683c9c7a0a :)